react/features/toolbox/components/web/Drawer.js

```diff
@@ -45,29 +45,27 @@ function Drawer({
     onClose,
     t }: Props) {
     const [ expanded, setExpanded ] = useState(false);
-    const drawerRef: Object = useRef(null);
 
     /**
-     * Closes the drawer when clicking or touching outside of it.
+     * Handles clicks within the menu, preventing the propagation of the click event.
      *
-     * @param {Event} event - Mouse down/start touch event object.
+     * @param {Object} event - The click event.
      * @returns {void}
      */
-    function handleOutsideClickOrTouch(event: Event) {
-        if (drawerRef.current && !drawerRef.current.contains(event.target)) {
-            onClose();
-        }
+    function handleInsideClick(event) {
+        event.stopPropagation();
     }
 
-    useEffect(() => {
-        window.addEventListener('mousedown', handleOutsideClickOrTouch);
-        window.addEventListener('touchstart', handleOutsideClickOrTouch);
-
-        return () => {
-            window.removeEventListener('mousedown', handleOutsideClickOrTouch);
-            window.removeEventListener('touchstart', handleOutsideClickOrTouch);
-        };
-    }, [ drawerRef ]);
+    /**
+     * Handles clicks outside of the menu, closing it, and also stopping further propagation.
+     *
+     * @param {Object} event - The click event.
+     * @returns {void}
+     */
+    function handleOutsideClick(event) {
```

Comment:
Why do we need this function to be defined in the component? Isn't it going to be recreated on every rerender? It looks like it doesn't use anything from the component itself.

Reply:
Fixed the redefinitions. (A more lengthy answer above :D)